code/modules/mob/living/simple_animal/bot/honkbot.dm

```diff
@@ -0,0 +1,398 @@
+
+
+/mob/living/simple_animal/bot/honkbot/hitby(atom/movable/AM, skipcatch = 0, hitpush = 1, blocked = 0)
+	if(istype(AM, /obj/item))
+		playsound(loc, honksound, 50, 1, -1)
+		var/obj/item/I = AM
+		if(I.throwforce < src.health && I.thrownby && (istype(I.thrownby, /mob/living/carbon/human)))
+			var/mob/living/carbon/human/H = I.thrownby
+			retaliate(H)
+	..()
+
+/mob/living/simple_animal/bot/honkbot/proc/bike_horn() //use bike_horn
+	if (emagged <= 1)
+		if(!client) //check if a player is controlling
+			playsound(loc, honksound, 50, 1, -1)
+		else
+			if (!spam_flag)
+				playsound(loc, honksound, 50, 1, -1)
+				spam_flag = 1 //prevent spam
+		icon_state = "honkbot-c"
+		spawn(5)
+			icon_state = "honkbot[on]"
+		spawn(cooldowntimehorn)
+			spam_flag = 0
+
+	else if (emagged == 2) //emagged honkbots will spam short and memorable sounds.
+
+		if (!client)
+			playsound(loc, pick('sound/items/bikehorn.ogg', 'sound/items/AirHorn2.ogg', 'sound/misc/sadtrombone.ogg', 'sound/items/AirHorn.ogg', 'sound/effects/reee.ogg',  'sound/items/WEEOO1.ogg', 'sound/voice/biamthelaw.ogg', 'sound/voice/bcreep.ogg','sound/magic/Fireball.ogg' ,'sound/effects/pray.ogg', 'sound/voice/hiss1.ogg','sound/machines/buzz-sigh.ogg', 'sound/machines/ping.ogg', 'sound/weapons/flashbang.ogg', 'sound/weapons/bladeslice.ogg'), 50, 0)
+			// to be put at 100 volume: (Weeoo1, bcreep, blaw, hiss, flashbang )
+		else
+			if (!spam_flag)
+				playsound(loc, pick('sound/items/bikehorn.ogg', 'sound/items/AirHorn2.ogg', 'sound/misc/sadtrombone.ogg', 'sound/items/AirHorn.ogg', 'sound/effects/reee.ogg', 'sound/items/WEEOO1.ogg', 'sound/voice/biamthelaw.ogg', 'sound/voice/bcreep.ogg','sound/magic/Fireball.ogg' ,'sound/effects/pray.ogg', 'sound/voice/hiss1.ogg','sound/machines/buzz-sigh.ogg', 'sound/machines/ping.ogg', 'sound/weapons/flashbang.ogg', 'sound/weapons/bladeslice.ogg'), 50, 0)
+				spam_flag = 1 // prevent spam
+		icon_state = "honkbot-e"
+		spawn(30) // keep flashing
+			icon_state = "honkbot[on]"
+		spawn(cooldowntimehorn)
+			spam_flag = 0
+
+/mob/living/simple_animal/bot/honkbot/proc/honk_attack(mob/living/carbon/C) // horn attack
+	if (!client) //check if a player is controlling
+		playsound(loc, honksound, 50, 1, -1)
+	else
+		if(spam_flag == 0)
+			playsound(loc, honksound, 50, 1, -1)
+			spam_flag = 1 // prevent spam
+	icon_state = "honkbot-c"
+	spawn(5)
+		icon_state = "honkbot[on]"
+	spawn(cooldowntimehorn)
+		spam_flag = 0
+
+/mob/living/simple_animal/bot/honkbot/proc/stun_attack(mob/living/carbon/C) // airhorn stun
+	if (!client)
+		playsound(loc, 'sound/items/AirHorn.ogg', 100, 1, -1) //HOOOOOOOOOOOOONK!!
+	else
+		if(spam_flag == 0)
+			playsound(loc, 'sound/items/AirHorn.ogg', 100, 1, -1) //HEEEEEEEEEEEENK!!
+	icon_state = "honkbot-c"
+	spawn(5)
+		icon_state = "honkbot[on]"
+	if(spam_flag == 0)
+		if(ishuman(C))
+			C.stuttering = 20
+			C.Jitter(50)
+			C.Knockdown(80)
+			var/mob/living/carbon/human/H = C
+			if (ckey != null) //don't alter AI behavior
```

Comment:
if(ckey)

Reply:
if you ghost with if(ckey), it just makes it braindead.
With (ckey != null), they return to their AI behavior.